docs/reference/contributing/target/sai.md

```diff
@@ -0,0 +1,78 @@
+<h2 id="sai-port">Serial Audio Interface (SAI)</h2>
+
+Implementing SAI enables Mbed OS to emit and receive an audio data stream.
+
+<span class="warnings">**Warning:** We are changing the SAI API in an upcoming release of Mbed OS. This page documents code that exists on a feature branch of Mbed OS. You can find details on how it may affect you in the [implementing the RTC API](#implementing-the-rtc-api) section.
+
+### Assumptions
+
+#### Defined behavior
+
+- `sai_init()` returns `SAI_RESULT_INVALID_PARAM` if at least one of the given parameters is undefined (NULL).
```

Comment:
`STATUS` vs `RESULT` ? What the rest of the HAL returns? I recall some did prexix_status_t

do we expect to check this in each HAL implementation or above layer should handle this checking? I would vote for above (we had this before in some HAL)

Reply:
I choose result because it's the result of the `sai_init` operation and not a poll-able `status` of the peripheral.
We expect the hal API to validate its input and the layer above to respect the API requirements and defined behaviours.
In a typical use case, a the layer above shouldn't have to check for null pointers etc as it is providing these objects.

Reply:
👍

Reply:
Regarding return types, please make sure its consistent (we might want to review all feature branches that are adding new HAL). Some might have status in there, or other.